File: 162-find-peak-element/find-peak-element.js
var findPeakElement = function(nums) {
    let i = 0
    let j = nums.length - 1

    while (i <= j) {
        let mid = Math.floor((i + j) / 2)

        if (mid > 0 && nums[mid] < nums[mid - 1]) j = mid - 1
        else if (mid + 1 < nums.length && nums[mid] < nums[mid + 1]) i = mid + 1
        else return mid

    }    

    return -1
};

File: 162-find-peak-element/find-peak-element.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        i = 0
        j = len(nums) - 1

        while (i <= j):
            mid = (i + j) // 2

            if (mid > 0 and nums[mid] < nums[mid - 1]):
                j = mid - 1
            elif (mid < len(nums) - 1 and nums[mid] < nums[mid + 1]):
                i = mid + 1
            else:
                return mid



        